crates/bevy_ecs/macros/src/lib.rs

```diff
@@ -446,26 +451,42 @@ pub fn derive_system_label(input: TokenStream) -> TokenStream {
     trait_path
         .segments
         .push(format_ident!("SystemLabel").into());
-    derive_label(input, &trait_path, "system_label")
+    let mut id_path = bevy_ecs_path();
+    id_path.segments.push(format_ident!("schedule").into());
+    id_path.segments.push(format_ident!("SystemLabelId").into());
+    derive_label(input, &trait_path, &id_path, "system_label")
 }
 
 /// Generates an impl of the `StageLabel` trait.
 ///
-/// This works only for unit structs, or enums with only unit variants.
-/// You may force a struct or variant to behave as if it were fieldless with `#[stage_label(ignore_fields)]`.
+/// For unit structs and enums with only unit variants, a cheap implementation can easily be created.
+///
+/// More complex types must be boxed and interned
+/// - opt in to this by annotating the entire item with `#[stage_label(intern)]`.
+///
+/// Alternatively, you may force a struct or variant to behave as if
+/// it were fieldless with `#[stage_label(ignore_fields)]`.
```

Comment:
```suggestion
/// Alternatively, you may force a struct or variant to behave as if
/// it were fieldless with `#[stage_label(ignore_fields)]`.
/// This is especially useful for [`PhantomData`](core::marker::PhantomData) fields.
```
`s/especially useful/only intended/` to taste